crates/bevy_ptr/src/lib.rs

```diff
@@ -176,6 +176,20 @@ impl<'a> PtrMut<'a> {
     pub fn as_ptr(&self) -> *mut u8 {
         self.0.as_ptr()
     }
+
+    /// Gets a `PtrMut` from this with a smaller lifetime
+    #[inline]
+    pub fn shrink(&mut self) -> PtrMut<'_> {
```

Comment:
If this method is chained together as part of a complex, I feel like it would be very unclear what this method is actually doing. `reborrow` would be a better name IMO.